### PR TITLE
formulae_dependents: refactor to use `bottled?` test-bot method

### PR DESCRIPTION
**Pull request by carlocab**
We already use the `bottled?` method from `test-bot` elsewhere, so it
seems better to be consistent about it. While we're doing that, let's
make sure to handle `:all` bottles appropriately.

### CODE REVIEW

File: lib/tests/formulae_dependents.rb [hunk 1]
```diff
@@ -29,15 +29,11 @@ def dependent_formulae!(formula_name, args:)
...
-          # TODO: A better thing to do here would be to attempt a source build
-          #       of unbottled dependents, but reverse the failure condition; i.e.
-          #       a successful source build is a failure, but a failed source build
-          #       is a pass to enable detection of unbottled formulae that should be bottled.
-          next unless bottled
+          # TODO: Work out how to use this code to identify
+          #       dependents that can be bottled. See
+          #       https://github.com/Homebrew/homebrew-test-bot/pull/678
+          next unless bottled?(dependent, no_older_versions: true)
+          next if bottled?(dependent, :all) && dependent.deps.any? { |d| !bottled?(d.to_formula) }
```
Comment (carlocab) [inline lib/tests/formulae_dependents.rb:36]: Unsure if I should be passing `no_older_versions: true` here too, as in
```suggestion
          next if bottled?(dependent, :all) && dependent.deps.any? { |d| !bottled?(d.to_formula, no_older_versions: true) }
```

File: lib/tests/formulae_dependents.rb [hunk 2]
```diff
@@ -102,17 +98,12 @@ def dependents_for_formula(formula, formula_name, args:)
...
-            dependents.select { |dep| dep.bottled? && !dep.bottle_specification.tag?(:all) }
-          else
-            dependents.select(&:bottled?)
-          end
+        bottled_dependents = dependents.select do |dep|
+          # If a dependent has an all bottle, we need to check that the dependent's dependencies
+          # have useable bottles. Otherwise, we can check the dependent directly.
+          next bottled?(dep, no_older_versions: true) unless bottled?(dep, :all)
+
+          dep.deps.all? { |d| bottled?(d.to_formula) }
```
Comment (carlocab) [inline lib/tests/formulae_dependents.rb:106]: Same here.

File: lib/tests/formulae_dependents.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply lib/tests/formulae_dependents.rb:36]: Yeh, I reckon probably. Better to have missing tests here than false negatives/positives. Same below.

**Review by MikeMcQuaid — APPROVED** [review]

File: lib/tests/formulae_dependents.rb (hunk 1 above)
Comment (carlocab) [reply lib/tests/formulae_dependents.rb]: Done, however, I also enabled testing of formulae with `:all` bottles whose dependency was built on the current OS earlier in the `test-bot` run.

For example, some dependents of `mono` have an `:all` bottle, but `mono` has no Linux bottle. We probably want to check those `mono` dependents if/when `mono` is bottled on Linux in a `test-bot` run.

Comment (MikeMcQuaid) [reply lib/tests/formulae_dependents.rb]: Works for me 👍🏻

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**